### ISSUE TITLE
Highlighting again

### ISSUE THREAD

**Issue by agaida** [tsujan/FeatherPad#97]
Hi, i know i can be pita - but it would be nice to have a default highlighter active if highlighting is activated - ok, i can choose 'url' every time i have a simple text that is not recognized, new or a paste - but it still sucks to be honest. 

Btw: I like url for that really much - urls are highlighted, whitespaces and line endings too - really nice.

**Comment by tsujan** [comment]
Sorry, I didn't understand it. Please be more specific and explain it with an example.

A highlighter (`QSyntaxHighlighter`) takes CPU time. It processes every character of every line and scans the text line-by-line downwards and then, re-scans it again whenever it changes. Do you mean an optional `URL` highlighter with normal texts?

Needless to say, I'm open to all suggestions, especially yours, which have lifted FeatherPad to a higher level.

**Comment by agaida** [comment]
ok, no problem - will do a little screencast

**Comment by tsujan** [comment]
> shouldn't it be highlighted? (From the video)

Yes, it should. I forgot emails in URL files (*.url).; will add it.

So, you want URL highlighting with normal texts and enabled with new tabs -- it'll format whitespaces automatically. I'd guessed right ;) It can be added as an option.

As usual, I'll think about the implementation and will tell you about the result.

**Comment by agaida** [comment]
cool, thanks

**Comment by tsujan** [comment]
Thank *you* for these good ideas. You might ask why they didn't occur to me. The answer is simple: because of the way I used text editors. Now, with these features, I use FP differently :)

**Comment by tsujan** [comment]
Done in https://github.com/tsujan/FeatherPad/commit/7c200d389ec4b39f80fae05f2c41c0fca5d4e0df

Now, whitespaces, line/doc ends and position lines are shown in normal texts too. If the user enables one of them, that means he wants it everywhere. Also, hyperlinks will be highlighted and parentheses will be matched automatically in that case.

So, there was no need to a separate option; enable at least one of them and normal texts will have a minimal highlighter.

I also added a "Normal" item to the end of the language menu after a separator because I felt angry about its absence ;)

This is still experimental. Contrary to its simple appearance, its code is complex and I had to solve several problems in it -- hence the delay. So, lots of tests are needed before the next release.
